src/widget/woverview.cpp

```diff
@@ -93,9 +93,9 @@ void WOverview::setup(const QDomNode& node, const SkinContext& context) {
     m_marks.setup(m_group, node, context, m_signalColors);
 
     for (int i = 0; i < m_marks.size(); ++i) {
-        WaveformMark& mark = m_marks[i];
-        if (mark.m_pPointCos) {
-            mark.m_pPointCos->connectValueChanged(this,
+        WaveformMark* mark = m_marks[i].data();
```

Comment:
Use QSharedPointer::data() sparsely, because it is not compatible with std::shared_pointer and switching from QSharedPointer to std::shared_pointer in the future becomes harder.

A const-ref is almost as efficient here and does not modify the reference count internally:
const QSharedPointer<WaveformMark>& pMark = m_marks[i];

Btw: Declaring a typedef WaveformMarkPointer for QSharedPointer<WaveformMark> could help to make the code more readable.